src/core/thread/data_poll_manager.cpp

```diff
@@ -170,6 +170,23 @@ void DataPollManager::SetExternalPollPeriod(uint32_t aPeriod)
     }
 }
 
+uint32_t DataPollManager::GetKeepAlivePollPeriod(void)
+{
+    uint32_t pollperiod = 0;
+
+    if (mExternalPollPeriod != 0)
+    {
+        pollperiod = mExternalPollPeriod;
+    }
+    else
+    {
+        pollperiod = Timer::SecToMsec(GetMeshForwarder().GetNetif().GetMle().GetTimeout()) -
```

Comment:
This code is duplicated from `DataPollManager::CalculatePollPeriod()`.  Shall we separate this computation out into a new method like `DataPollManager::GetDefaultPollPeriod()`?